scripts/src/lib/generator/TemplateGenerator.ts

```diff
@@ -0,0 +1,75 @@
+import JSZip from 'jszip';
+import FileSaver from 'file-saver';
+import { addGradleWrapper } from './GradleWrapper';
+import { addGroovyGradle } from './GradleGroovy';
+import { getKotlinAdapterVersions, getLoaderVersions, getMinecraftYarnVersions } from '../Api';
+import { addModJson } from './ModJson';
+
+export interface Options {
+    minecraftVersion: string,
+    projectName: string,
+    packageName: string,
+    useKotlin: boolean,
+    dataGeneration: boolean,
+    githubActions: boolean,
+    splitSources: boolean,
+}
+
+export interface KotlinOptions {
+	kotlinVersion: string,
+	fabricKotlinAdapterVersion: string
+}
+
+// Computed options are not presented to the user.
+export interface ComputedOptions extends Options {
+	modid: string,
+	loaderVersion: string,
+	yarnVersion: string,
+	kotlin: KotlinOptions | undefined
+}
+
+export interface TemplateOptions {
+	minecraftVersion: string, loaderVersion: string, yarnVersion: string
+}
+
+export async function generateTemplate(options: Options) {
+	const computedOptions = await computeOptions(options);
+	const zip = new JSZip();
+
+	await addGradleWrapper(zip);
+	addGroovyGradle(zip, computedOptions);
+	addModJson(zip, computedOptions);
+
+	zip.generateAsync({ type: 'blob' }).then((content) => {
```

Comment:
If it does get extracted out into a library then this probably shouldn't work on just a zip basis, it'd make a lot more sense to have some interface for writing to either the filesystem or a zip file.